File: Backend/js/vaoSetup.js
let triangleVertices = new Float32Array ([
    // Position     // Color
    0.0,  0.5, 0.0, 1.0, 0.0, 0.0,
   -0.5, -0.5, 0.0, 0.0, 1.0, 0.0,
    0.5, -0.5, 0.0, 0.0, 0.0, 1.0
]);

const vao = gl.createVertexArray();
const vbo = gl.createBuffer();

// Bind vao and vbo
gl.bindVertexArray(vao);
gl.bindBuffer(gl.ARRAY_BUFFER, vbo);

// Set up vbo data
gl.bufferData(gl.ARRAY_BUFFER, triangleVertices, gl.STATIC_DRAW);
            
// layout(location = 0) in vec3 aPos
gl.vertexAttribPointer(0, 3, gl.FLOAT, false, 6 * 4, 0);
gl.enableVertexAttribArray(0);

// layout(location = 1) in vec3 aColor;
gl.vertexAttribPointer(1, 3, gl.FLOAT, false, 6 * 4, 12)
gl.enableVertexAttribArray(1);

File: Backend/cpp/vaoSetup.cpp
float triangleVertices[] = {
    // Position     // Color
    0.0,  0.5, 0.0, 1.0, 0.0, 0.0,
   -0.5, -0.5, 0.0, 0.0, 1.0, 0.0,
    0.5, -0.5, 0.0, 0.0, 0.0, 1.0
};

uint32_t vao, vbo;
glGenVertexArrays(1, &vao);
glGenBuffers(1, &vbo);

// Bind vao and vbo
glBindVertexArray(vao);
glBindBuffer(GL_ARRAY_BUFFER, vbo);

// Set up vbo data
glBufferData(GL_ARRAY_BUFFER, sizeof(triangleVertices), triangleVertices, GL_STATIC_DRAW);

// layout(location = 0) in vec3 aPos
glVertexAttribPointer(0, 3, GL_FLOAT, false, 6 * sizeof(float), 0);
glEnableVertexAttribArray(0);

// layout(location = 1) in vec3 aColor;
glVertexAttribPointer(1, 3, GL_FLOAT, false, 6 * sizeof(float), (void*)(3 * sizeof(float)));
glEnableVertexAttribArray(1);